tikv/server.go

```diff
@@ -194,16 +192,27 @@ func (p *kvScanProcessor) Process(key, value []byte) (err error) {
 func (svr *Server) KvPrewrite(ctx context.Context, req *kvrpcpb.PrewriteRequest) (*kvrpcpb.PrewriteResponse, error) {
 	reqCtx, err := newRequestCtx(svr, req.Context, "KvPrewrite")
 	if err != nil {
-		return &kvrpcpb.PrewriteResponse{Errors: convertToKeyErrors([]error{err})}, nil
+		return &kvrpcpb.PrewriteResponse{Errors: []*kvrpcpb.KeyError{convertToKeyError(err)}}, nil
 	}
 	defer reqCtx.finish()
 	if reqCtx.regErr != nil {
 		return &kvrpcpb.PrewriteResponse{RegionError: reqCtx.regErr}, nil
 	}
 	errs := svr.mvccStore.Prewrite(reqCtx, req.Mutations, req.PrimaryLock, req.GetStartVersion(), req.GetLockTtl())
-	return &kvrpcpb.PrewriteResponse{
-		Errors: convertToKeyErrors(errs),
-	}, nil
+
+	resp := new(kvrpcpb.PrewriteResponse)
+	var keyErrors []*kvrpcpb.KeyError
+	for _, err := range errs {
+		if err != nil {
+			if regErr := extractRegionError(err); regErr != nil {
```

Comment:
There is too much boilerplate code to set region error and key error.
We can make the function return both region error and key error/errors.
Like:
```
resp.RegionError, resp.Errors = convertPBErrors(errs)
```

```
resp.RegionError, resp.Error = convertToPBError(err)
```